packages/fluxible-plugin-devtools/src/lib/devtools-plugin.js

```diff
@@ -0,0 +1,169 @@
+             * @method overridectx
+             * @return {void}
+             */
+            function overrideCtx() {
+                debug('devtools plugin %s', enableDebug ? 'enabled' : 'disabled')
+                if(!enableDebug) {
+                    return;
+                }
+                /**
+                 * Override the _createSubActionContext method so we can track "parent" actions
+                 * for each executed action. We can later use this to graph our the dependencies.
+                 */
+                const createSubActionContext = context._createSubActionContext.bind(context);
+                context._createSubActionContext = function createDevSubActionContext(parentActionContext, action) {
+                    let subActionContext = createSubActionContext(parentActionContext, action);
+                    let actionReference = {
+                        rootId: subActionContext.rootId,
+                        name: subActionContext.displayName
+                    };
+                    if (!parentActionContext.__parentAction) {
+                        // new top level action
+                        actionReference.type = (typeof window === 'undefined') ? 'server' : 'client';
+                        actionHistory.push(actionReference);
+                    } else {
+                        // append child action
+                        const parent = parentActionContext.__parentAction;
+                        parent.children = parent.children || [];
+                        parent.children.push(actionReference);
+                    }
+                    // for extablishing parent->child relationships
+                    subActionContext.__parentAction = actionReference;
+                    // for updating start/end times of the actionReference
+                    action.__actionReference = actionReference;
+                    return subActionContext;
+                }
+            }
+            overrideCtx();
+
+            return {
+                /**
+                 * Modyfy and return new arguments to `callAction` in FluxibleContext
+                 * @method plugExecuteAction
+                 * @param {Object} actionContext the current action context
+                 * @param {Function} action action An action creator function that receives actionContext, payload,
+                 *  and done as parameters
+                 * @param {Object} payload The action payload
+                 * @param {Function} done Method to be called once action execution has completed
+                 */
+                plugExecuteAction: function plugExecuteAction(options) {
+                    if (!enableDebug) {
+                        // unchanged
+                        return options;
+                    }
+
+                    const action = options.action;
+                    const actionReference = action.__actionReference
```

Comment:
I don't think it's safe to attach on to the `action` in this way in case there are multiple executions of the same action at the same time.